src/lifecycle/renderer.js

```diff
@@ -1,20 +1,18 @@
 import fragment from '../api/fragment';
 
-const defaultRenderer = function (elem, render) {
+function defaultRenderer (elem, opts) {
   while (elem.childNodes.length) {
     elem.removeChild(elem.childNodes[0]);
   }
-  elem.appendChild(fragment(render()));
-};
-
-export default function renderer (opts) {
-  let render = opts.render ? opts.render : null;
-  let renderer = opts.renderer ? opts.renderer.bind(opts) : defaultRenderer;
-  let resolvedAttribute = opts.resolvedAttribute;
+  elem.appendChild(fragment(opts.render(elem)));
```

Comment:
If you were to return only text from your `render()` method, then this default renderer would fail because `fragment()` didn't accept text nodes.